torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -11,27 +11,72 @@
 namespace torch {
 namespace nn {
 
-EmbeddingOptions::EmbeddingOptions(int64_t count, int64_t dimension)
-    : count_(count), dimension_(dimension) {}
+EmbeddingOptions::EmbeddingOptions(int64_t num_embeddings, int64_t embedding_dim) : num_embeddings(num_embeddings), embedding_dim(embedding_dim) {}
 
-EmbeddingImpl::EmbeddingImpl(EmbeddingOptions options) : options(options) {
-  reset();
+EmbeddingImpl::EmbeddingImpl(EmbeddingOptions options): options(options) {
+  EmbeddingImpl::reset();
 }
 
 void EmbeddingImpl::reset() {
-  weight = register_parameter(
-      "weight", torch::empty({options.count_, options.dimension_}));
-  NoGradGuard guard;
-  weight.normal_(0, 1);
+  if (options.padding_idx() != c10::nullopt){
+    if(*options.padding_idx() > 0){
+      assert((*options.padding_idx() < *options.num_embeddings()) && "Padding_idx must be within num_embeddings");
+    }
+    else{
```

Comment:
This is not strictly equivalent to the Python implementation - the Python side reads:
```python
elif padding_idx < 0:
```
which is not the same here